### PR TITLE
fix(common/core/web): Remove empty rows in OSK 🍒 

### PR DESCRIPTION
**Pull request by darcywong00**
🍒 pick of #5699 and ed490fcbeffa12ecbd9dc7fdd7a42e35e5d6c720 to stable-14.0 in addressing the KeymanWeb portion of #5327

If a touch keyboard had empty rows (compiled with older versions of Keyman Developer), the OSK would generate exceptions.
This adds a fixed version of `ActiveLayout.correctLayerEmptyRowBug()` that removes the empty rows.

For testing, I checked in the afghan_turkmen keyboard as included on the issue. I also created a custom `empty_row` keyboard in Developer 13 and edited the js file so there's an empty row in each of the 3 layers (different row ID).

## User Testing

* **TEST_ANDROID** Verifies OSK handles empty rows in the embedded app
Use an Android 5.0 emulator so the polyfill code is exercised.
1. Load the PR build of Keyman for Android
2. Test several of the default sil_euro_latin keys and verify they function correctly
3. Download and install [afghan_turkmen.kmp](https://darcywong00.github.io/examples/afghan/afghan_turkmen.kmp) which has an empty top row defined on the default layer
4. After the keyboard finished installing and switches, verify the afghan_turkmen OSK is fine (no empty rows)
5. Verify the default keys work. Note: since the attached keyboard came from the community site, it has an issue where the shift button doesn't switch layers

* **TEST_WEB** Verifies OSK 
Use the CI web test page artifacts using the new page `testing/empty-row/` in mobile viewport
1. Load the test page
2. With the empty_row keyboard, verify the OSK on the default layer is fine and keys function correctly
3. Use the shift to switch to the shift layer
4. Verify the shift layer OSK is fine and keys function correctly
5. Use the numeric key to switch to the numeric layer
6. Verify the numeric layer OSK is fine and keys function correctly
7. Use the globe key to switch to the afghan_turkmen keyboard
8. Verify the default layer on the afghan_turkmen OSK is fine (no empty rows)
9. Verify the keys work (except for the shift button, see the mentioned limitation in TEST_ANDROID)

### CODE REVIEW

File: web/source/osk/visualKeyboard.ts [hunk 1]
```diff
@@ -863,25 +863,6 @@ namespace com.keyman.osk {
...
-            break;
-          }
-        }
-
-        if(i < rows.length) {
-          rows.splice(i-rows.length,rows.length-i);
-        }
-      }
-      // ...remove to here when compiler bug fixed ***
-
```
Comment (darcywong00) [inline web/source/osk/visualKeyboard.ts:884]: Corresponds to https://github.com/keymanapp/keyman/commit/ed490fcbeffa12ecbd9dc7fdd7a42e35e5d6c720#diff-5480cfedac96cf36181f5fdcdd51aee3e44404c27c923584fceb8da14fde9b6cL43-L58

**Review by jahorton — APPROVED** [review]
LGTM pending user testing.

**Comment by MakaraSok** [comment]
* **TEST_ANDROID: PASSED** everything works well as indicated.
   


* **TEST_WEB: PASSED** tested on Chrome emulator (iPhone 6/7/8)
   * the default, shift and numeric layers of `empty_row` work fine.
   * no empty row seen on `afghan_turkmen` keyboard and the keyboard works fine except the one caveat pointed out in the instructions.

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 14.0.281](https://keyman.com/downloads/releases/stable/14.0.281)

**Pull request merged**